### PR TITLE
Update node documentation to better explain the access token mechanism

### PR DESCRIPTION
**Pull request by vanruch**

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -25,10 +26,20 @@ The majority of errors along to the error code provide additional information in
 }
 ```
 
-## Group Token
-
-Some requests require an access token for authorization.
+## Group Token [Token]
 
+Some requests like account management or accessing to restricted data fields require an access token for authorization. The access token is a base64-encoded JSON file of following format:
```
Comment (spherefoundry) [inline apiary.apib]: We should also add that we are using a deterministic way to serialise the JSON into a string. This should probably be explained in detail in a separate section, with a link here.

File: apiary.apib [hunk 2]
```diff
@@ -11,6 +11,7 @@ Before you can start interacting with API you need to create an account. An acco
 Each state-modifying call needs to have the `content.idData.createdBy` field with creator address and one of two:
 * A body field `content.signature`. It is generated with ethereum sign function from the serialized `content.idData` object. Serialisation algorithm will be described in separate section.
 * A http authorization header `AMB ...`: You can supply your secret to API and gateway will calculate the signature for you. __Warning__: This is only allowed in development and testing environment. We strongly recommend against using it in a production environment.
+* A http authorization header `AMB_TOKEN ...`: see more in [Token section](Token)
```
Comment (spherefoundry) [inline apiary.apib]: This should be above the DEV header as it's a regular vs. exception mechanism.

File: apiary.apib [hunk 3]
```diff
@@ -8,23 +8,25 @@ Welcome to Ambrosus Node API Documentation.
...
+Depending on the type of operation you want to perform you are required to provide one of two forms of authentication: signed content or token based.
+
+For entity upload operations the body of the request needs to contain a `content.signature` field. It is generated by signing the [serialized](Serialization) `content.idData` object using the standard ethereum sign function and your private key.
+
+For other operations (modifying accounts and querying entities) a token should be provided using the http authorization header with `AMB_TOKEN ...` . The calculation of a token if described in the [Token section](Token). 
+
+There is a special third option where you provide a http authorization header `AMB ...` with your account private key. When doing so the gateway will calculate the signature for you. This mechanism can be enabled using a config parameter. __Warning__: Sharing your private key is a security risk. This mechanism should therefor only be used on development and testing environments.
```
Comment (vanruch) [inline apiary.apib]: development -> the development

File: apiary.apib [hunk 4]
```diff
@@ -8,23 +8,25 @@ Welcome to Ambrosus Node API Documentation.
...
-* A body field `content.signature`. It is generated with ethereum sign function from the [serialized](Serialization) `content.idData` object.
-* A http authorization header `AMB_TOKEN ...`: see more in [Token section](Token)
-* A http authorization header `AMB ...`: You can supply your secret to API and gateway will calculate the signature for you. __Warning__: This is only allowed in development and testing environment. We strongly recommend against using it in a production environment.
+Depending on the type of operation you want to perform you are required to provide one of two forms of authentication: signed content or token based.
+
+For entity upload operations the body of the request needs to contain a `content.signature` field. It is generated by signing the [serialized](Serialization) `content.idData` object using the standard ethereum sign function and your private key.
+
+For other operations (modifying accounts and querying entities) a token should be provided using the http authorization header with `AMB_TOKEN ...` . The calculation of a token if described in the [Token section](Token).
```
Comment (vanruch) [inline apiary.apib]: The calculation of a token if -> is

File: apiary.apib [hunk 5]
```diff
@@ -8,14 +8,30 @@ Welcome to Ambrosus Node API Documentation.
...
+Depending on the type of operation you want to perform you are required to provide one of two forms of authentication: signed content or token based.
+
+For entity upload operations the body of the request needs to contain a `content.signature` field. It is generated by signing the [serialized](Serialization) `content.idData` object using the standard ethereum sign function and your private key.
+
+For other operations (modifying accounts and querying entities) a token should be provided using the http authorization header with `AMB_TOKEN ...` . The calculation of a token if described in the [Token section](Token). 
+
+There is a special third option where you provide a http authorization header `AMB ...` with your account private key. When doing so the gateway will calculate the signature for you. This mechanism can be enabled using a config parameter. __Warning__: Sharing your private key is a security risk. This mechanism should therefor only be used on development and testing environments.
```
Comment (Anze1m) [inline apiary.apib]: therefor -> therefore

File: apiary.apib (hunk 3 above)
Comment (vanruch) [reply apiary.apib]: therefor -> therefore

Comment (vanruch) [reply apiary.apib]: used in development and testing environments

File: apiary.apib [hunk 6]
```diff
@@ -8,14 +8,30 @@ Welcome to Ambrosus Node API Documentation.
...
+
+When serialising data for hashing and signing, special care needs to be taken so that the result is deterministic. Otherwise the output values may not match, and lead to validation errors. Following rules need to be followed:
+
+* The JSON is serialized to a single-line string without any whitespace characters outside of strings
+* All fields in objects are sorted lexicographically
+* In other aspects the output should match `JSON.stringify`
+
+## Hashing
+
+Hashes are a very important part of proving the persistence and the immutability of the data. All hashes in the AMB-NODE are computed using the [Keccak256](https://keccak.team/index.html) algorithm.
```
Comment (Anze1m) [inline apiary.apib]: I wouldn't say that, as we are not using pure keccak.
https://web3js.readthedocs.io/en/1.0/web3-eth-accounts.html#hashmessage

**Pull request merged**